test/atat-iam.test.ts

```diff
@@ -23,3 +23,212 @@ describe("ATAT network creation", () => {
     }
   });
 });
+
+describe("ATAT IAM Policy creation", () => {
+  let app: cdk.App;
+  let stack: cdk.Stack;
+  let template: Template;
+
+  beforeEach(() => {
+    app = new cdk.App();
+    stack = new AtatIam.AtatIamStack(app, "TestIamStack", {});
+    template = Template.fromStack(stack);
+  });
+
+  test("Fully assert generalReadAccess IAM managed policy with matchers", async () => {
+    template.hasResourceProperties(
+      "AWS::IAM::ManagedPolicy",
+      Match.objectEquals({
+        PolicyDocument: {
+          Statement: [
+            {
+              Action: ["dynamodb:*GetItem", "dynamodb:PartiQLSelect", "dynamodb:Scan", "dynamodb:Query"],
+              Effect: "Allow",
+              Resource: {
+                "Fn::Join": [
+                  "",
+                  [
+                    "arn:",
+                    {
+                      Ref: "AWS::Partition",
+                    },
+                    ":dynamodb:*:",
+                    {
+                      Ref: "AWS::AccountId",
+                    },
+                    ":table/*",
+                  ],
+                ],
+              },
+              Sid: "DynamoDBItemAccess",
+            },
+            {
+              Action: "apigateway:GET",
+              Effect: "Allow",
+              Resource: {
+                "Fn::Join": [
+                  "",
+                  [
+                    "arn:",
+                    {
+                      Ref: "AWS::Partition",
+                    },
+                    ":apigateway:*::/restapis*",
+                  ],
+                ],
+              },
+              Sid: "APIGatewayRestApiReadAccess",
+            },
+            {
+              Action: ["states:List*", "states:Describe*", "states:GetExecutionHistory"],
+              Effect: "Allow",
+              Resource: {
+                "Fn::Join": [
+                  "",
+                  [
+                    "arn:",
+                    {
+                      Ref: "AWS::Partition",
+                    },
+                    ":states:*:",
+                    {
+                      Ref: "AWS::AccountId",
+                    },
+                    ":stateMachine:*",
+                  ],
+                ],
+              },
+              Sid: "StepFunctionsReadAccess",
+            },
+            {
+              Action: ["xray:BatchGetTraces", "xray:Get*", "xray:List*"],
+              Effect: "Allow",
+              Resource: "*",
+              Sid: "XRayReadAccess",
+            },
+          ],
+          Version: "2012-10-17",
+        },
+        Description: "Grants read access to specific resources not in ViewOnlyAccess",
+        Path: "/",
+      })
+    );
```

Comment:
I am about to open a PR that actually removes at the `dynamodb:` permissions. The others will probably stay. I am not sure that the individual permissions that we associate with each policy are necessarily worth testing. I think there _are_ something we want to test:

- Named exports (which are basically the stack's API) exist as expected and the ARNs parse to a resource of the correct type
- That specific _named_ IAM roles/policies exist (anything where the `ResourceName` attribute, such as `roleName` or `policyName` is provided)
- We don't create any resources of type `AWS::IAM::Policy` (this may fail right now...)

Reply:
I agree re: testing the individual permissions but the APIGW one looks great. I think the place where we'll see the most value is in testing our own utility methods to ensure that they do what they intend to do.

Reply:
The PR has been updated with the proper IAM policies in https://github.com/dod-ccpo/atat-web-api/pull/641/commits/5793aa47f8d88c36774ff4b9f2fc46b6d6e7bed8. The `AWS::IAM::Policy` tests are not failing, you can see in the checks for the PR that all the jest tests completed.

Reply:
We are already testing the named exports for `roleName`, but I am unsure why we wouldn't want to test the actual policy statement to ensure it is what we expect?

Reply:
My initial issue was that it wasn't clear that we were actually testing our own code with a `given -> when -> expect` structure but upon further review I think they are actually good insofar as we don't just slap whatever our templates generate into those tests but treat those statements as something that we maintain over time with an eye on what we really want to see there.